### PR TITLE
Add JSON support

### PR DESCRIPTION
**Pull request by tdryer**
This adds a `json` feature based on `serde_json` exposing `Request::json` and `Response::with_json`.

Are you interested in this or a similar approach for adding JSON support to vial?

### CODE REVIEW

**Comment by xvxx** [comment]
Wow, very cool! And yes, I am indeed interested in this direction!

I think of Vial as "the à la carte micro micro-framework." So my only request here is that this feature be named "serde_json" or something else that makes it less generic, so that we can make room for additional "JSON" features.

For example, lately I have been using [tinyjson](https://crates.io/crates/tinyjson) - it's got zero dependencies and it's really easy to use, but it's different from Serde. I can easily see myself using Serde in one Vial project, and tinyjson in another.

That said, I like the `json` and `with_json` methods. So I think it's okay to call the feature something specific like "serde_json", but leave the method names as generic. If someone wants to use both tinyjson and Serde in a project, they always have the option of including the deps manually in their `Cargo.toml`. 

What do you think? And thanks again!

**Comment by tdryer** [comment]
Sounds good. I renamed the feature to `json_serde` since features and dependencies cannot have the same name.

File: docs/index.html [hunk 1]
```diff
@@ -505,27 +505,27 @@ <h4 id="a-micro-micro-framework">~ a micro micro-framework ~</h4>
 <h2 id="hello-world">Hello World</h2>
 <p>Here’s the bare minimum:</p>
 <div class="sourceCode" id="cb1"><pre class="sourceCode rust"><code class="sourceCode rust"><span id="cb1-1"><a href="#cb1-1"></a><span class="pp">vial::routes!</span> <span class="op">{</span></span>
-<span id="cb1-2"><a href="#cb1-2"></a>    GET <span class="st">&quot;/&quot;</span> =&gt; |_| <span class="st">&quot;Greetings, creature.&quot;</span>;</span>
+<span id="cb1-2"><a href="#cb1-2"></a>    GET <span class="st">&quot;/&quot;</span> <span class="op">=&gt;</span> <span class="op">|</span>_<span class="op">|</span> <span class="st">&quot;Greetings, creature.&quot;</span><span class="op">;</span></span>
```
Comment (tdryer) [inline docs/index.html:508]: Most of this diff is from rendering with a different version of pandoc.

Comment (xvxx) [reply docs/index.html:508]: No problem, I might just make the docs generation into a GH:Action so we don't have to play wack-a-mole with non-cargo dev dependency versioning. Either that or modify it to use syntect + pulldown-cmark so everyone *is* on the same version. I'm not sure that "just install whatever and then run this random Ruby in the Makefile" is setting future generations of contributors up for success...

**Comment by xvxx** [comment]
Awesome, thank you! And thanks for adding the tests. I just released **v0.1.5** with this included.

**Pull request merged**